judge/forms.py

```diff
@@ -95,6 +97,26 @@ def __init__(self, *args, **kwargs):
         self.fields['user_script'].widget = AceWidget(mode='javascript', theme=user.profile.resolved_ace_theme)
 
 
+class EmailChangeForm(Form):
+    password = CharField(widget=forms.PasswordInput())
+    email = forms.EmailField()
+
+    def __init__(self, *args, user, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.user = user
+
+    def clean_email(self):
+        if User.objects.filter(email=self.cleaned_data['email']).exists():
```

Comment:
What about deactivated accounts?

Reply:
I believe the current check (during registration) includes all users, including deactivated ones. Should we allow users to reuse emails from deactivated emails?

Reply:
I think a better solution might be to set the emails of deactivated accounts to a dummy email and leave the current checks intact, cc @quantum5 @Xyene for thoughts.

Reply:
Agree, I think this mirroring the registration logic is good.